presto-main/src/main/java/io/prestosql/execution/QueryStateTimer.java

```diff
@@ -87,25 +93,27 @@ private void beginDispatching(long now)
 
     public void beginPlanning()
     {
-        beginPlanning(tickerNanos());
+        beginPlanning(tickerNanos(), currentThreadCpuTime());
     }
 
-    private void beginPlanning(long now)
+    private void beginPlanning(long now, long cpuNow)
     {
         beginDispatching(now);
         dispatchingTime.compareAndSet(null, nanosSince(beginDispatchingNanos, now));
         beginPlanningNanos.compareAndSet(null, now);
+        beginPlanningCpuNanos.compareAndSet(null, cpuNow);
     }
 
     public void beginStarting()
     {
-        beginStarting(tickerNanos());
+        beginStarting(tickerNanos(), currentThreadCpuTime());
     }
 
-    private void beginStarting(long now)
+    private void beginStarting(long now, long cpuNow)
     {
-        beginPlanning(now);
+        beginPlanning(now, cpuNow);
         planningTime.compareAndSet(null, nanosSince(beginPlanningNanos, now));
+        planningCpuTime.compareAndSet(null, nanosSince(beginPlanningCpuNanos, cpuNow));
```

Comment:
could you assert that thread id is the same in `beginStarting` `beginPlanning`?

Reply:
Thanks! Addressed!